### PR TITLE
Add healthcheck to create-container

### PR DESCRIPTION
**Pull request by bubaflub**
Builds on work from PR #1152.  Differences include:
- Resolved the merge conflict
- pulled out the healthcheck into a separate file in docker/types
- added an integration test that exercises a healthcheck

I'm open to any suggestions to clean this up.

### CODE REVIEW

**Comment by bubaflub** [comment]
Also including @ryan0x44 who helped me with this and @shin- who we've chatted with briefly about this.

File: docker/api/container.py [hunk 1]
```diff
@@ -320,7 +321,7 @@ def start(self, container, binds=None, port_bindings=None, lxc_conf=None,
               dns=None, dns_search=None, volumes_from=None, network_mode=None,
               restart_policy=None, cap_add=None, cap_drop=None, devices=None,
               extra_hosts=None, read_only=None, pid_mode=None, ipc_mode=None,
-              security_opt=None, ulimits=None):
+              security_opt=None, ulimits=None, healthcheck=None):
```
Comment (shin-) [inline docker/api/container.py]: This part is incorrect - we do not want to update the `start` method with new parameters. Especially in this case, healthcheck is part of the `ContainerConfig`, not the `HostConfig`, so it's not valid to have here.

File: docker/utils/utils.py [hunk 2]
```diff
@@ -1025,6 +1025,11 @@ def create_container_config(
             'stop_signal was only introduced in API version 1.21'
         )
 
+    if healthcheck is not None and compare_version('1.24', version) < 0:
```
Comment (shin-) [inline docker/utils/utils.py]: You can use `version_gte(version, '1.24')` here, which is more explicit.

File: docker/api/container.py (hunk 1 above)
Comment (shin-) [inline docker/api/container.py]: If you fix that it should solve the issue with the tests as well.

**Comment by bubaflub** [comment]
Great, I'll fix things up and get the tests to pass.

**Comment by bubaflub** [comment]
I've rebased off of master and fixed the merge conflict.  I've made the changes you requested and now all of `make test` passes locally.  I also appeased pep8.  Slightly related -- I used autopep8 to fix up those nits.  Highly recommended.

**Comment by shin-** [comment]
LGTM now, thanks!

**Comment by shin-** [comment]
Merged via #1294

**Pull request closed without merging**